supportedBackends/google/pipelines/common/src/main/scala/cromwell/backend/google/pipelines/common/authentication/PipelinesApiVMAuthentication.scala

```diff
@@ -34,19 +35,24 @@ object PipelinesApiDockerCredentials {
 
   def apply(dockerCredentials: DockerCredentials, googleConfig: GoogleConfiguration): PipelinesApiDockerCredentials = {
     // If there's an encryption key defined there must be a valid auth defined to encrypt it.
-    val credential = dockerCredentials.keyName flatMap { _ =>
-      val authName = dockerCredentials.authName.getOrElse(throw new RuntimeException("KMS Encryption key defined for private Docker but no auth specified"))
-      googleConfig.auth(authName) match {
-        case Invalid(e) => throw new RuntimeException("Error creating GoogleAuthMode: " + e.toList.mkString(", "))
-        case Valid(auth) => auth.apiClientGoogleCredential(identity)
-      }
+    val authValidation = dockerCredentials.keyName match {
```

Comment:
ToL I hear the cool kids are using `fold` instead of pattern matching.  Something about less code

Reply:
OTOH Odersky has stated his disapproval with the cool kids so YMMV ;)

Reply:
No danger of me being mistaken for a cool kid. Anyway I thought we collectively decided to generally avoid `fold`ing `Option`s due to the impact on readability?